crypto/evp/digest.c

```diff
@@ -170,8 +170,15 @@ int EVP_DigestInit_ex(EVP_MD_CTX *ctx, const EVP_MD *type, ENGINE *impl)
         ctx->provctx = NULL;
     }
 
-    if (type != NULL)
+    if (type != NULL) {
         ctx->reqdigest = type;
+    } else {
+        if (!ctx->digest) {
```

Comment:
== NULL? (Yes I know you just moved it :))

Reply:
Fixed